### ISSUE TITLE
Increase memlock resource limit for PID 1

### ISSUE THREAD

**Issue by jkartseva** [systemd/systemd#15053]
systemd sets 64M as the hard limit for memlock for PID 1 [[1]](https://github.com/systemd/systemd/blob/master/src/basic/def.h#L60). Locked memory is required for BPF program to be loaded, e.g. firewall program with any of `IPAddressAllow`, `IPAddressDeny` options or device control w/ `DevicePolicy ` option.
The underlying kernel mechanism is that locked memory is accounted per user but limited per process [[2]](https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/tree/kernel/bpf/syscall.c#n343)   so if there is a process consuming large amount of memory other processes running under the same user (root) may fail to lock the required amount of memory.
 As BPF become more widespread, this becoming an issue for PID 1 having too low limit of 64M.

E.g.:
```
/bin/systemd-run --unit run-testcommand -p DevicePolicy=strict sh -c 'echo hello > /dev/null'
run-testcommand.service: Failed at step STDIN spawning /bin/sh: Operation not permitted
```
A flexible solution would be to make memlock rlimit configurable, e.g. via env var or command line options.
Since this may not be the case for PID 1 as was discussed in #9205, changing the hardcoded `HIGH_RLIMIT_MEMLOCK` from 64M to e.g. 512M would also work.
Another approach is to lock `x`, then if `x` is not enough to load a prog and `-EPERM` is returned, try locking `2x` until a max is reached.
Couple that with verboase warning in journal.

[1] https://github.com/systemd/systemd/blob/master/src/basic/def.h#L60
[2] https://git.kernel.org/pub/scm/linux/kernel/git/bpf/bpf-next.git/tree/kernel/bpf/syscall.c#n343

**Comment by poettering** [comment]
Hmm, I'd prefer not to make this configurable for now. It's a safety limit only after all. Hence let's find some scheme to make it work for everyone. i.e. high enough to not be hit in real-life, but low enough to provide a safety margin if things go wrong. hence I propose to scale the limit by physical RAM size.

**Comment by poettering** [comment]
My proposed fix is in #15551. It set the limit to a third of physical RAM, which I think should be an OK value to start from. Let's then see where this goes.
